### PR TITLE
Ci/lint english

### PR DESCRIPTION
**Pull request by hleonardtek**
Implement an english linter to check the spelling of the whitepaper files.

- Full errors location described in the action logs
- Report sent by mail and commented in the PR
- Easy custom with dictionaries to enhance vocabulary

### CODE REVIEW

**Review by ccamel — CHANGES_REQUESTED** [review]

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  workflow_dispatch:
+
+jobs:
+  spelling:
+    name: Spell checking
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - uses: check-spelling/check-spelling@main
```
Comment (ad2ien) [inline .github/workflows/spellcheck.yml:13]: Souldn't we settle the version?
```suggestion
      - uses: check-spelling/check-spelling@v0.0.20-alpha5
```

File: .github/workflows/spellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        id: spelling
+        with:
+          experimental_apply_changes_via_bot: 1
+          experimental_path: docs/whitepaper
+          extra_dictionaries:
```
Comment (ad2ien) [inline .github/workflows/spellcheck.yml:18]: What happens if we remove these?

**Review by ad2ien — CHANGES_REQUESTED** [review]
Nice thanks, maybe let's just fix the version.

File: .github/workflows/spellcheck.yml (hunk 2 above)
Comment (hleonardtek) [reply .github/workflows/spellcheck.yml:18]: Effectively we can remove the `experimental_apply_changes_via_bot`.
However, `experimental_path` is used to check only certain files (here the whitepaper folder), and `extra_dictionaries` is used for external dictionaries. If we remove them we will have to build our own dictionary for basic stuff (words such as "http", "https", etc)

File: .github/workflows/spellcheck.yml (hunk 1 above)
Comment (hleonardtek) [reply .github/workflows/spellcheck.yml:13]: I will put the last one, i.e v0.0.20-alpha7.
Thanks !

**Review by ad2ien — APPROVED** [review]
Ok for me!

File: .github/workflows/spellcheck.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  spelling:
+    name: Spell checking
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - uses: check-spelling/check-spelling@v0.0.20-alpha7
+        id: spelling
+        with:
+          experimental_path: docs/whitepaper
```
Comment (ad2ien) [inline .github/workflows/spellcheck.yml:16]: @hleonardtek actually Anik should be the one commenting the PR

```suggestion
          GITHUB_TOKEN: ${{ secrets.OKP4_TOKEN }}
```

**Comment by amimart** [comment]
Regarding the comments it did on this PR, I'm not really convinced this linter will be helpful.. Maybe a cron job opening a new PR with spelling and grammar suggestions would fit much, what do you think ?

**Comment by ccamel** [comment]
If you don't mind, I put the PR in draft while we agree on the best approach. Thx.

**Comment by ad2ien** [comment]
> Regarding the comments it did on this PR, I'm not really convinced this linter will be helpful.. Maybe a cron job opening a new PR with spelling and grammar suggestions would fit much, what do you think ?

I though it was nice to have a comment on the PR, it gives people a chance to correct their typos. Maybe with a better template?
Opening a PR without changes? an issue maybe then?

**Comment by hleonardtek** [comment]
Yes effectively,  like I wrote in the issue, this action doesn't provide relevant information concerning the errors. It only prints the misspelled words in the PR, without the concerned files and precise locations.
However, we could build our own action, or even a CJ as said by @amimart.
In my view, an action is more suitable as it will be triggered on changes only.

What are the advantages of a CronJob compared to an action ?

**Comment by ccamel** [comment]
@amimart still relevant?

**Comment by amimart** [comment]
@ccamel Would have been great, but seems difficult to continue so I'm in favour of closing it.

**Pull request closed without merging**